src/encoding.rs

```diff
@@ -832,9 +903,9 @@ pub mod message {
         B: Buf,
     {
         check_wire_type(WireType::LengthDelimited, wire_type)?;
-        let mut msg = M::default();
-        merge(WireType::LengthDelimited, &mut msg, buf)?;
-        messages.push(msg);
+        let len = messages.len();
+        messages.resize_with(len + 1, M::default);
```

Comment:
nice, there are probably more opportunities in the codebase for improvements like this.  With prost's model of inlining message fields, resulting in large structs, these can be a big win.